packages/manager/core/application/src/hooks/useFeatureAvailability.ts

```diff
@@ -0,0 +1,35 @@
+import { useQuery } from '@tanstack/react-query';
+import { apiClient } from '@ovh-ux/manager-core-api';
+
+export const featureAvailabilityBaseQueryKey = 'feature-availability-';
+
+/**
+ * @example
+ * const {data, error, loading} = useFeatureAvailability({applicationName: 'dedicated-nasha', featureList: ['feature1', 'feature2']});
+ * const isAppAvailable = data.['dedicated-nasha'];
+ * const isFeature1Available = data.feature1;
+ * const isFeature2Available = data.feature2;
+ */
+export const useFeatureAvailability = ({
+  applicationName,
+  featureList = [],
+}: {
+  applicationName: string;
+  featureList?: string[];
+}) => {
+  const featuresListToCheck = [applicationName].concat(
+    featureList.map((name) => `${applicationName}:${name}`),
+  );
+
+  const fetchFeatureAvailabilityData = async () => {
+    const result = await apiClient.aapi.get(
+      `/feature/${featuresListToCheck.join(',')}/availability`,
+    );
+    return result.data || {};
+  };
+
+  return useQuery<Record<string, boolean>>(
+    [`${featureAvailabilityBaseQueryKey}-${featuresListToCheck.join('-')}`],
+    fetchFeatureAvailabilityData,
+  );
```

Comment:
**nitpick:** We need not limit the feature-flipping to query related to one particular app. A query can contain features from multiple apps.

Reply:
Yes it is currently a possibility
But, except in the container, is it a good idea to be able to add coupling between different apps ?
I think it doesn't make sense that the availability of a feature in a particular app impacts another app.

Additionally, the calls are cached in react-query so, even if the container has to make multiple calls, the apps using the hook will be able to use the result directly, instead of doing a feature availability call again when opening the app

Reply:
@chipp972, I think this is the feature we can keep because we can have multiple products depending on each other (especially on the PCI, for example, Instance/Floating IP/Gateway/Private Network/Load Balancer are related to each other and we may be in a situation to use this.) More over API doesn't restrict us to have only `json` for each product. So just for Baremetal server, you can have multiple json grouped based on any particular features.

Reply:
Ok then I put the possibility to passe multiple objects to create one request